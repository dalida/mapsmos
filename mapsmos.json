{
    "META" : {
        "MIR_CLF33A":{
            "mode":"LAND",
            "title":"Soil Moisture",
            "copyright" : "Copyright IFREMER, ESA,...",
            "param_id":"Soil_Moisture",
            "inc":null,
            "nbColors":20,
            "vmin":0.0,
            "vmax":0.5,
            "size":[1280,720]
        },
        "MIR_CSF3AA":{
            "mode":"SEA",
            "title":"Mean Sea Surface Salinity",
            "copyright" : "Copyright IFREMER, ESA,...",
            "param_id":"Mean_Sea_Surface_Salinity",
            "inc":null,
            "nbColors":20,
            "vmin":32.0,
            "vmax":38.0,
            "size":[1280,720]
        }
    },

    "MODES" : {
        "SEA":{
            "lon_0":-50.0,
            "mapargs":{"projection":"moll", "resolution":"l", "lon_0":-50.0, "ellps":"WGS84", "anchor":"S"},
            "colormap":"jet",
            "fill_continents":true
        },
        "LAND":{
            "lon_0":0.0,
            "mapargs":{},
            "colormap":"hot",
            "fill_continents":false
        }
    }
}