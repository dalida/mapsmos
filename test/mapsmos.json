{
    "META" : {
        "MIR_CLF33A":{
            "mode":"LAND",
            "title":"Soil Moisture",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Soil_Moisture",
            "inc":null,
            "nbColors":20,
            "vmin":0.0,
            "vmax":0.5,
            "size":[1280,720]
        },
        "MIR_CSF3AA":{
            "mode":"SEA",
            "title":"Mean Sea Surface Salinity",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Mean_Sea_Surface_Salinity",
            "inc":null,
            "nbColors":20,
            "vmin":32.0,
            "vmax":38.0,
            "size":[1280,720]
        },
        "MIR_CSF3AD":{
            "mode":"SEA",
            "title":"Mean Sea Surface Salinity",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Mean_Sea_Surface_Salinity",
            "inc":null,
            "nbColors":20,
            "vmin":32.0,
            "vmax":38.0,
            "size":[1280,720]
        },
        "MIR_CLF31A":{
            "mode":"LAND",
            "title":"Soil Moisture",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Soil_Moisture",
            "inc":null,
            "nbColors":20,
            "vmin":0.0,
            "vmax":0.5,
            "size":[1280,720]
        },
        "MIR_CDF3RA":{
            "mode":"LAND",
            "title":"Probability RFI",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Prob_RFI",
            "inc":10,
            "nbColors":20,
            "vmin":0.0,
            "vmax":0.5,
            "size":[1280,720]
        },
        "MIR_CDF3RA":{
            "mode":"LAND",
            "title":"RFI Contamination Probability ",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Prob_RFI",
            "inc":10,
            "nbColors":20,
            "vmin":0.0,
            "vmax":0.5,
            "size":[1280,720]
        },
        "MIR_CDF3TA":{
            "mode":"LAND",
            "title":"Brightness Temperature H-pol",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"BT_H",
            "inc":10,
            "nbColors":20,
            "vmin":0.0,
            "vmax":0.5,
            "size":[1280,720]
        },
        "MIR_CSF2PA":{
            "mode":"SEA",
            "title":"Sea Surface Salinity",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Sea_Surface_Salinity_Model1_Value",
            "inc":null,
            "nbColors":20,
            "vmin":32.0,
            "vmax":38.0,
            "size":[1280,720]
        },
        "MIR_CLF3EA":{
            "mode":"LAND",
            "title":"Dielectric Constant from MD",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Dielect_Const_Module",
            "inc":null,
            "nbColors":20,
            "vmin":0.0,
            "vmax":0.5,
            "size":[1280,720]
        },
        "MIR_CLF3DA":{
            "mode":"LAND",
            "title":"Retrieved Soil Moisture Value (med, min, max)",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Soil_Moisture",
            "inc":null,
            "nbColors":20,
            "vmin":0.0,
            "vmax":0.5,
            "size":[1280,720]
        },
        "MIR_CLF3MA":{
            "mode":"LAND",
            "title":"Retrieved Soil Moisture Value",
            "copyright" : "Copyright Ifremer, ESA,...",
            "param_id":"Soil_Moisture",
            "inc":null,
            "nbColors":20,
            "vmin":0.0,
            "vmax":0.5,
            "size":[1280,720]
        }
    },

    "MODES" : {
        "SEA":{
            "lon_0":-50.0,
            "mapargs":{"projection":"moll", "resolution":"l", "lon_0":-50.0, "ellps":"WGS84", "anchor":"S"},
            "colormap":"jet",
            "fill_continents":true
        },
        "LAND":{
            "lon_0":0.0,
            "mapargs":{},
            "colormap":"hot",
            "fill_continents":false
        }
    }
}
