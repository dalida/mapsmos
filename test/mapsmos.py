#!/usr/bin/env python 
# -*- coding: utf-8 -*-

import argparse
import json
import os

from dateutil import parser as dateparser
from matplotlib.colors import Normalize
from mpl_toolkits.basemap import Basemap
import netCDF4

import matplotlib.cm as cm
import matplotlib.pyplot as plt
import numpy as np


DPI = 100.

def fromto(fname):
    deb = dateparser.parse(fname.split("_")[4])
    fin = dateparser.parse(fname.split("_")[5])
    fmt = '%Y/%m/%d %H:%M'
    return deb.strftime(fmt) + ' to ' + fin.strftime(fmt)


def guess_product_type(product_name):
    ll = product_name.split('_')
    return '_'.join(ll[2:4])


def read_product_conf(ptype, path=None):
    if not path:
        path = os.path.dirname(__file__) + '/mapsmos.json'
    with open(path) as f:
        conf = json.load(f)
    meta = conf['META'][ptype]
    mode = conf['MODES'][meta['mode']]
    return meta, mode


def mapsmos(smos_file, output=None, conf=None, cut=None):
    """ SMOS products map generator.
      
        Use mapsmos.json to configure each product types
        
        see http://matplotlib.org/basemap/api/basemap_api.html#mpl_toolkits.basemap.Basemap
        for cartographic projection settings (i.e. --mapargs-- property in mapsmos.json)
        
    """
    fig_file = os.path.basename(smos_file).split('.')[0] + '.png'

    if output:
        fig_file = output + '/' + os.path.basename(fig_file)
    ptype = guess_product_type(os.path.basename(smos_file))
    meta, mode = read_product_conf(ptype, conf)
    # lecture du fichier
    ncsmos = netCDF4.Dataset(smos_file)
    latsmos = ncsmos.variables['lat'][:]
    lonsmos = ncsmos.variables['lon'][:]
    param = ncsmos.variables[meta['param_id']]
    if meta['inc'] is not None:
        ndparam = param[meta['inc'], :, :]
    else:
        ndparam = param[:, :]
    if cut is not None:
        meta['vmin'], meta['vmax'] = np.nanpercentile(ndparam.filled(np.nan), [cut, 100-cut])
    title = meta['title'] + ', ' + fromto(os.path.basename(smos_file))
    lonsmos = np.fmod(lonsmos+180.+3600.0, 360.) - 180.
    index = np.argsort(lonsmos)
    lonsmos = lonsmos[index].flatten()
    latsmos = latsmos.flatten()
    ndparam = ndparam[:, index]
    # visualisation graphique
    width, height = meta["size"]
    fig = plt.figure(figsize=(width/DPI, height/DPI), dpi=DPI)
    ax = fig.add_axes([0.1,0.1,0.8,0.8])
    #ax.set_title(title, fontdict={'fontsize': 20.0,  'verticalalignment': 'baseline', })
    m = Basemap(**mode['mapargs'])
    m.drawcoastlines(linewidth=0.01, antialiased=False)
    m.drawmapboundary(fill_color='white', linewidth=0.01)
    m.drawmeridians(np.arange(-180, 181, 60),
        labels=[0, 0, 0, 0],
        #linewidth=0.8,
        labelstyle=None, )
    m.drawparallels(np.arange(-90, 91, 30),
        labels=[1, 0, 0, 0],
        #linewidth=0.8,
        labelstyle=None, )
    if mode['fill_continents']:
        m.fillcontinents(color='grey')
    ticks = np.linspace(meta['vmin'], meta['vmax'] , meta['nbColors']/2+1)
    lonsmosout, z = m.shiftdata(lonsmos, ndparam, lon_0=mode['lon_0'])
    lon, lat = np.meshgrid(lonsmosout, latsmos)
    x, y = m(lon, lat)
    cmap = cm.get_cmap(mode['colormap'], meta['nbColors'])
    cmap.set_bad('1.0')
    #cmap.set_over((0.0, 0.0, 1.0, 1.0))
    #cmap.set_under((0.0, 1.0, 0.0, 1.0))
    pc = m.pcolormesh(x, y, z, norm=Normalize(meta['vmin'], meta['vmax']), cmap=cmap)
    cb = plt.colorbar(pc,
                shrink=0.8,
                orientation='horizontal',
                fraction=0.04,
                extend='both',
                ticks=ticks,
                pad=0.08)
    cb.set_label(param.long_name + ' (%s)' % (param.units))
    plt.title(title, fontdict={'fontsize': 18.0,  'verticalalignment': 'baseline', })
    plt.text(1.0, -0.03, meta['copyright'], 
                horizontalalignment='right',
                verticalalignment='center',
                transform=ax.transAxes)
    plt.savefig(fig_file)
    plt.close()
    ncsmos.close()


if __name__ == '__main__':
    parser = argparse.ArgumentParser()
    parser.add_argument("filepath", type=str, help="SMOS Product full path")
    parser.add_argument("-o", "--output", default=None, help="Output directory")
    parser.add_argument("-c", "--conf", default=None, help="Mapsmos configuration file path")
    args = parser.parse_args()
    mapsmos(args.filepath, output=args.output, conf=args.conf)