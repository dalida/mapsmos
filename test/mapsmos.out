Nettoyage des cartes
diss_path : /home/catds_diss/data/cpdc/vqual/current/MAP
LOOP through MAP tree
ftp_dir : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF33A/2016/08
file to match : SM_OPER_MIR_CLF33A_20160801T000000_20160803T235959_300_001_7
ftp_dir : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF31A/2016/214
file to match : SM_OPER_MIR_CLF31A_20160801T000000_20160801T235959_300_001_7
ftp_dir : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3EA/2016/08
file to match : SM_OPER_MIR_CLF3EA_20160801T000000_20160803T235959_300_001_7
ftp_dir : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AA/2016/08
file to match : SM_OPER_MIR_CSF3AA_20160801T000000_20160831T235959_300_001_7
ftp_dir : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AD/2016/08
file to match : SM_OPER_MIR_CSF3AD_20160801T000000_20160831T235959_300_001_7
ftp_dir : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF2PA/2016/214
file to match : SM_OPER_MIR_CSF2PA_20160801T000000_20160801T235959_300_001_7
ftp_dir : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3RA/2016/214
file to match : SM_OPER_MIR_CDF3RA_20160801T000000_20160830T235959_310_001_7
ftp_dir : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3TA/2016/214
file to match : SM_OPER_MIR_CDF3TA_20160801T000000_20160801T235959_310_001_7
Generation des cartes
root : /home/catds_diss/data/cpdc/vqual/current/FTP
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3RA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3RA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3RA/2016/214
filename : SM_OPER_MIR_CDF3RA_20160801T000000_20160830T235959_310_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/SMOS/GRIDDED/L3/OPER/MIR_CDF3RA/2016/214
file_to_match : SM_OPER_MIR_CDF3RA_20160801T000000_20160830T235959_310_001_7
La carte associee au produit SM_OPER_MIR_CDF3RA_20160801T000000_20160830T235959_310_001_7.tgz existe deja
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3RD
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3RD/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3RD/2016/214
filename : SM_OPER_MIR_CDF3RD_20160801T000000_20160830T235959_310_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3TA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3TA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3TA/2016/214
filename : SM_OPER_MIR_CDF3TA_20160801T000000_20160801T235959_310_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/SMOS/GRIDDED/L3/OPER/MIR_CDF3TA/2016/214
file_to_match : SM_OPER_MIR_CDF3TA_20160801T000000_20160801T235959_310_001_7
La carte associee au produit SM_OPER_MIR_CDF3TA_20160801T000000_20160801T235959_310_001_7.tgz existe deja
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3TD
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3TD/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SMOS/GRIDDED/L3/OPER/MIR_CDF3TD/2016/214
filename : SM_OPER_MIR_CDF3TD_20160801T000000_20160801T235959_310_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF33A
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF33A/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF33A/2016/08
filename : SM_OPER_MIR_CLF33A_20160801T000000_20160803T235959_300_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/SM/GRIDDED/L3SM/OPER/MIR_CLF33A/2016/08
file_to_match : SM_OPER_MIR_CLF33A_20160801T000000_20160803T235959_300_001_7
La carte associee au produit SM_OPER_MIR_CLF33A_20160801T000000_20160803T235959_300_001_7.tgz existe deja
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF31A
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF31A/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF31A/2016/214
filename : SM_OPER_MIR_CLF31A_20160801T000000_20160801T235959_300_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/SM/GRIDDED/L3SM/OPER/MIR_CLF31A/2016/214
file_to_match : SM_OPER_MIR_CLF31A_20160801T000000_20160801T235959_300_001_7
La carte associee au produit SM_OPER_MIR_CLF31A_20160801T000000_20160801T235959_300_001_7.tgz existe deja
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF31D
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF31D/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF31D/2016/214
filename : SM_OPER_MIR_CLF31D_20160801T000000_20160801T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF33D
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF33D/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF33D/2016/08
filename : SM_OPER_MIR_CLF33D_20160801T000000_20160803T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3ED
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3ED/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3ED/2016/08
filename : SM_OPER_MIR_CLF3ED_20160801T000000_20160803T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3EA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3EA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3EA/2016/08
filename : SM_OPER_MIR_CLF3EA_20160801T000000_20160803T235959_300_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/SM/GRIDDED/L3SM/OPER/MIR_CLF3EA/2016/08
file_to_match : SM_OPER_MIR_CLF3EA_20160801T000000_20160803T235959_300_001_7
La carte associee au produit SM_OPER_MIR_CLF3EA_20160801T000000_20160803T235959_300_001_7.tgz existe deja
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3DD
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3DD/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3DD/2016/08
filename : SM_OPER_MIR_CLF3DD_20160801T000000_20160810T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3DA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3DA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3DA/2016/08
filename : SM_OPER_MIR_CLF3DA_20160801T000000_20160810T235959_300_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/SM/GRIDDED/L3SM/OPER/MIR_CLF3DA/2016/08
file_to_match : SM_OPER_MIR_CLF3DA_20160801T000000_20160810T235959_300_001_7
map_file is None...
file to process...
echec de la generation de la carte a partir du produit SM_OPER_MIR_CLF3DA_20160801T000000_20160810T235959_300_001_7.DBL.nc avec le code erreur 1-d or 2-d data required
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3MA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3MA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3MA/2016/08
filename : SM_OPER_MIR_CLF3MA_20160801T000000_20160831T235959_300_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/SM/GRIDDED/L3SM/OPER/MIR_CLF3MA/2016/08
map_file is None...
file to process...
succes de la generation de la carte a partir du produit SM_OPER_MIR_CLF3MA_20160801T000000_20160831T235959_300_001_7.DBL.nc
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3MD
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3MD/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/SM/GRIDDED/L3SM/OPER/MIR_CLF3MD/2016/08
filename : SM_OPER_MIR_CLF3MD_20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF2PA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF2PA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF2PA/2016/214
filename : SM_OPER_MIR_CSF2PA_20160801T000000_20160801T235959_300_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/OS/GRIDDED/L3OS/OPER/MIR_CSF2PA/2016/214
file_to_match : SM_OPER_MIR_CSF2PA_20160801T000000_20160801T235959_300_001_7
La carte associee au produit SM_OPER_MIR_CSF2PA_20160801T000000_20160801T235959_300_001_7.tgz existe deja
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF2PD
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF2PD/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF2PD/2016/214
filename : SM_OPER_MIR_CSF2PD_20160801T000000_20160801T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3DD
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3DD/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3DD/2016/08
filename : SM_OPER_MIR_CSF3DD_20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3BA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3BA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3BA/2016/08
filename : SM_OPER_MIR_CSF3BA_20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3B_
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3B_/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3B_/2016/08
filename : SM_OPER_MIR_CSF3B__20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3CD
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3CD/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3CD/2016/08
filename : SM_OPER_MIR_CSF3CD_20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3DA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3DA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3DA/2016/08
filename : SM_OPER_MIR_CSF3DA_20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3A_
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3A_/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3A_/2016/08
filename : SM_OPER_MIR_CSF3A__20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AA/2016/08
filename : SM_OPER_MIR_CSF3AA_20160801T000000_20160831T235959_300_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AA/2016/08
file_to_match : SM_OPER_MIR_CSF3AA_20160801T000000_20160831T235959_300_001_7
La carte associee au produit SM_OPER_MIR_CSF3AA_20160801T000000_20160831T235959_300_001_7.tgz existe deja
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3D_
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3D_/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3D_/2016/08
filename : SM_OPER_MIR_CSF3D__20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3BD
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3BD/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3BD/2016/08
filename : SM_OPER_MIR_CSF3BD_20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3C_
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3C_/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3C_/2016/08
filename : SM_OPER_MIR_CSF3C__20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3CA
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3CA/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3CA/2016/08
filename : SM_OPER_MIR_CSF3CA_20160801T000000_20160831T235959_300_001_7.tgz
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AD
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AD/2016
root : /home/catds_diss/data/cpdc/vqual/current/FTP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AD/2016/08
filename : SM_OPER_MIR_CSF3AD_20160801T000000_20160831T235959_300_001_7.tgz
map_dir : /home/catds_diss/data/cpdc/vqual/current/MAP/OS/GRIDDED/L3OS/OPER/MIR_CSF3AD/2016/08
file_to_match : SM_OPER_MIR_CSF3AD_20160801T000000_20160831T235959_300_001_7
La carte associee au produit SM_OPER_MIR_CSF3AD_20160801T000000_20160831T235959_300_001_7.tgz existe deja
Fin de la generation des cartes
